lib/hexpm_web/live/page_live.ex

```diff
@@ -0,0 +1,91 @@
+defmodule HexpmWeb.PageLive do
+  @moduledoc false
+  use HexpmWeb, :live_view
+
+  alias Hexpm.Repository.{Packages, Releases, Repository}
+
+  require Logger
+
+  @max_matches 5
+  @matches_sort :recent_downloads
+  @query_size_limit_bytes 100
+
+  @impl true
+  def render(assigns) do
+    HexpmWeb.PageView.render("index.html", assigns)
+  end
+
+  @impl true
+  def mount(_params, _session, socket) do
+    hexpm = Repository.hexpm()
+
+    IO.inspect(socket.assigns, label: "Socket assigns: ")
+
+    repositories =
+      Hexpm.Accounts.Users.all_organizations(nil)
+      |> Enum.map(& &1.repository)
+
+    # Assign expensive, one-time calculations first
+    socket =
+      socket
+      |> assign(:num_packages, Packages.count())
+      |> assign(:num_releases, Releases.count())
+      |> assign(:package_top, Packages.top_downloads(hexpm, "recent", 8))
+      |> assign(:package_new, Packages.recent(hexpm, 10))
+      |> assign(:releases_new, Releases.recent(hexpm, 10))
+      |> assign(:total, Packages.total_downloads())
+      |> assign(:repositories, repositories)
+
+    {:ok,
+     assign(socket,
+       container: nil,
+       custom_flash: true,
+       hide_search: true,
+       query: nil,
+       result: nil,
+       loading: false,
+       matches: []
+     ),
+     temporary_assigns: [
+       num_packages: 0,
+       num_releases: 0,
+       package_top: [],
+       package_new: [],
+       total: 0
+     ]}
+  end
+
+  @impl true
+  def handle_event("suggest", %{"search" => query}, socket)
+      when byte_size(query) <= @query_size_limit_bytes do
+    Logger.debug("Got suggest event in #{__MODULE__} with query: #{query}")
+
+    results =
+      Packages.search(
+        socket.assigns.repositories,
+        0,
+        @max_matches,
+        Hexpm.Utils.parse_search(query),
+        @matches_sort,
+        nil
+      )
+
+    {:noreply, assign(socket, matches: results)}
+  end
+
+  @impl true
+  def handle_event("search", %{"search" => query}, socket)
+      when byte_size(query) <= @query_size_limit_bytes do
+    {:noreply,
+     socket
+     |> redirect(
+       to:
+         Routes.package_path(
+           socket,
+           :index,
+           search: query,
+           sort: "recent_downloads"
+         )
+     )}
```

Comment:
Avoid single function pipes:

```suggestion
     redirect(socket,
       to:
         Routes.package_path(
           socket,
           :index,
           search: query,
           sort: "recent_downloads"
         )
     )}
```

Reply:
Fixed.